src/auto_code_show_and_followups_messages.py

```diff
@@ -0,0 +1,125 @@
+        RQA_AND_FOLLOW_UP_KEYS.append(plan.raw_field)
+
+    SENT_ON_KEY = "sent_on"
+    NOISE_KEY = "noise"
+    ICR_MESSAGES_COUNT = 200
+    ICR_SEED = 0
+
+    def log_empty_string_stats(data, raw_fields):
+        for raw_field in raw_fields:
+            total_messages_count = 0
+            empty_string_messages_count = 0
+
+            for td in data:
+                if raw_field in td:
+                    total_messages_count += 1
+                    if td[raw_field] == "":
+                        empty_string_messages_count += 1
+
+            log.debug(f"{raw_field}: {empty_string_messages_count} messages were \"\", out "
+                      f"of {total_messages_count} total")
+
+    @classmethod
+    def auto_code_show_and_followups_messages(cls, user, data, icr_output_dir, coda_output_dir):
+        # Filter out test messages sent by AVF.
+        if not PipelineConfiguration.DEV_MODE:
+            data = MessageFilters.filter_test_messages(data)
+        
+        # Filter out runs sent outwith the project start and end dates
+        data = MessageFilters.filter_time_range(
+            data, cls.SENT_ON_KEY, PipelineConfiguration.PROJECT_START_DATE, PipelineConfiguration.PROJECT_END_DATE)
+        
+        # Filter for runs which don't contain a response to any week's question or the follow-up questions
+        data = MessageFilters.filter_empty_messages(data, cls.RQA_AND_FOLLOW_UP_KEYS)
+
+        # Tag RQA and follow up messages which are noise as being noise
+        for td in data:
+            is_noise = True
+            for rqa_key in cls.RQA_AND_FOLLOW_UP_KEYS:
+                if rqa_key in td and not somali.DemographicCleaner.is_noise(td[rqa_key], min_length=10):
+                    is_noise = False
+            td.append_data({cls.NOISE_KEY: is_noise}, Metadata(user, Metadata.get_call_location(), time.time()))
+        
+        #TODO Uncomment once channels.py is ready
+        #Label each message with channel keys
+        #Channels.set_channel_keys(user, data, cls.SENT_ON_KEY)
+
+        # Filter for messages which aren't noise (in order to export to Coda and export for ICR)
+        not_noise = MessageFilters.filter_noise(data, cls.NOISE_KEY, lambda x: x)
+
+        # Compute the number of RQA messages that were the empty string
+        log.debug("Counting the number of empty string messages for each raw radio show field...")
+        raw_rqa_fields = []
+        for plan in PipelineConfiguration.RQA_CODING_PLANS:
+            if plan.raw_field not in raw_rqa_fields:
+                raw_rqa_fields.append(plan.raw_field)
+        cls.log_empty_string_stats(data, raw_rqa_fields)
+
+        # Compute the number of follow up messages that were the empty string
+        log.debug("Counting the number of empty string messages for each follow up field...")
+        raw_follow_up_fields = []
+        for plan in PipelineConfiguration.FOLLOW_UP_CODING_PLANS:
```

Comment:
could probably have merged this loop with the RQA loop

Reply:
Merged thank you!